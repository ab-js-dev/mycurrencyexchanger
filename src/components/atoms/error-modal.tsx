import React from 'react'
import { Modal, View, Text, StyleSheet, TouchableOpacity } from 'react-native'

interface ErrorModalProps {
  showError: boolean
  errorMessage: string
  onClose: () => void // Function to close the modal
}

const ErrorModal: React.FC<ErrorModalProps> = ({ showError, errorMessage, onClose }) => {
  return (
    <Modal animationType="slide" transparent={true} visible={showError} onRequestClose={onClose}>
      <View style={styles.centeredView}>
        <View style={styles.modalView}>
          <Text style={styles.errorMessage}>{errorMessage}</Text>
          <TouchableOpacity onPress={onClose} style={styles.button}>
            <Text style={styles.buttonText}>Close</Text>
          </TouchableOpacity>
        </View>
      </View>
    </Modal>
  )
}

const styles = StyleSheet.create({
  centeredView: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 22,
    backgroundColor: 'rgba(0, 0, 0, 0.5)' // Semi-transparent background
  },
  modalView: {
    margin: 20,
    backgroundColor: 'white',
    borderRadius: 20,
    padding: 35,
    alignItems: 'center',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5
  },
  errorMessage: {
    marginBottom: 15,
    textAlign: 'center',
    color: 'red' // Color for the error message text
  },
  button: {
    borderRadius: 10,
    padding: 10,
    elevation: 2,
    width: 100,
    backgroundColor: '#6200EE'
  },
  buttonText: {
    color: 'white',
    fontWeight: 'bold',
    textAlign: 'center'
  }
})

export default ErrorModal
