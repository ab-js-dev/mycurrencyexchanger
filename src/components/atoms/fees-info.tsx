import React, { type ReactElement } from 'react'
import { View, Text, FlatList, StyleSheet } from 'react-native'
import { FEE_TIERS, type Tiers } from '@static/fees'

const renderItem = ({ item }: { item: Tiers }): ReactElement => (
  <View style={styles.itemContainer}>
    <Text style={styles.amountText}>AED {item.limit} and above</Text>
    <Text style={styles.feeText}>{(item.fee * 100).toFixed(2)}%</Text>
    <Text style={styles.amountText}>As Fee</Text>
  </View>
)
const FeesInformation = (): ReactElement => {
  return (
    <View style={styles.container}>
      <Text style={styles.header}>Fee Structure</Text>
      <FlatList
        data={FEE_TIERS}
        renderItem={renderItem}
        keyExtractor={(item, index) => index.toString()}
      />
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    alignSelf: 'flex-start'
  },
  header: {
    fontSize: 22,
    fontWeight: 'bold',
    color: '#727c97',
    marginBottom: 15
  },
  itemContainer: {
    backgroundColor: '#fff',
    borderRadius: 10,
    padding: 15,
    marginBottom: 10,
    flexDirection: 'row',
    gap: 10,
    justifyContent: 'space-between',
    alignItems: 'center'
  },
  amountText: {
    fontSize: 14,
    color: '#9fa8b9'
  },
  feeText: {
    fontSize: 12,
    fontWeight: 'bold',
    color: '#007AFF'
  }
})

export default FeesInformation
