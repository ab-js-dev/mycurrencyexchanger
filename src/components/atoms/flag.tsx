import React, { type ReactElement } from 'react'
import CountryFlag from 'react-native-country-flag'

export const Flag = (props: { countryCode: string }): ReactElement<{ countryCode: string }> => (
  <CountryFlag isoCode={props.countryCode} size={25} />
)
