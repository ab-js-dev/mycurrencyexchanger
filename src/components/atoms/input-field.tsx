import React, { useEffect } from 'react'
import { TextInput, StyleSheet, type TextInputProps } from 'react-native'

type InputFieldProps = {
  isEditable?: boolean
  calculatedValue?: string
  onAmountChange: (text: string) => void
} & TextInputProps

const InputField: React.FC<InputFieldProps> = ({
  calculatedValue,
  isEditable = true,
  onAmountChange
}) => {
  const [value, setValue] = React.useState<string>('')
  const onChange = (val: string): void => {
    onAmountChange(val)
    setValue(val)
  }

  useEffect(() => {
    setValue(calculatedValue ?? '')
  }, [calculatedValue])
  return (
    <TextInput
      editable={isEditable}
      style={styles.input}
      value={value}
      onChangeText={onChange}
      placeholder="0.00"
      keyboardType="decimal-pad"
      testID="inputField"
    />
  )
}

const styles = StyleSheet.create({
  input: {
    flex: 1,
    height: 70,
    paddingHorizontal: 10,
    fontSize: 18,
    fontWeight: 'bold',
    color: '#0D1F2D',
    textAlign: 'right'
  }
})

export default InputField
