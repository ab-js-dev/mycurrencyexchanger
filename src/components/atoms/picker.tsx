import React, { type ReactElement } from 'react'
import { getAllCountries } from '@utils/countries'
import { Picker } from '@react-native-picker/picker'

export const CountryPicker = (props: {
  selectedCountry: string
  setSelectedCountry: (countryCode: string) => void
}): ReactElement => {
  const { selectedCountry, setSelectedCountry } = props
  const countries = getAllCountries()

  return (
    <Picker
      testID="countryPicker"
      selectedValue={selectedCountry}
      onValueChange={itemValue => {
        setSelectedCountry(itemValue)
      }}>
      {countries.map(country => (
        <Picker.Item
          testID="picker-item"
          key={country.countryCode}
          label={country.currency}
          value={country.countryCode}
        />
      ))}
    </Picker>
  )
}
