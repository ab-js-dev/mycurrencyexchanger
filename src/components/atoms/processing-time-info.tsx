import React, { type ReactElement } from 'react'
import { View, Text, StyleSheet } from 'react-native'
import { FontAwesome5 } from '@expo/vector-icons'

const ProcessingTimeInfo = (): ReactElement => (
  <View style={styles.container} testID="ProcessingTimeInfo">
    <View style={styles.infoTextContianer}>
      <FontAwesome5 name="bolt" size={20} color="black" testID="process-time-icon" />
      <Text style={styles.infoText}>Processing time - 1 Hour</Text>
    </View>
    <Text style={styles.disclaimerText}>*Normal working hours & public holidays apply</Text>
  </View>
)

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center'
  },
  infoTextContianer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    gap: 10,
    marginBottom: 10
  },
  infoText: {
    fontSize: 16,
    fontWeight: 'bold',
    color: 'black'
  },
  disclaimerText: {
    fontSize: 14,
    color: '#bdbec3'
  }
})

export default ProcessingTimeInfo
