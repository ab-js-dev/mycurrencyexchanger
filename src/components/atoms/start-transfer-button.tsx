import React, { type ReactElement } from 'react'
import { StyleSheet, Text, TouchableOpacity } from 'react-native'

interface StartTransferButtonProps {
  onPress: () => void
  disabled?: boolean
}
const StartTransferButton = ({ onPress, disabled }: StartTransferButtonProps): ReactElement => (
  <TouchableOpacity
    disabled={disabled}
    style={[styles.button, disabled ?? false ? styles.disabled : styles.enabled]}
    onPress={onPress}
    testID="StartTransferButton">
    <Text style={styles.buttonText}>Start transfer</Text>
  </TouchableOpacity>
)

const styles = StyleSheet.create({
  button: {
    paddingVertical: 12,
    paddingHorizontal: 25,
    borderRadius: 10,
    alignItems: 'center',
    justifyContent: 'center',
    width: '100%'
  },
  enabled: {
    backgroundColor: '#4F8EF7'
  },
  disabled: {
    backgroundColor: '#BDBDBD'
  },
  buttonText: {
    color: 'white',
    fontSize: 18,
    fontWeight: 'bold'
  }
})

export default StartTransferButton
