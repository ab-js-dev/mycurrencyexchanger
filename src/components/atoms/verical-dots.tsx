import React from 'react'
import { TouchableOpacity, StyleSheet } from 'react-native'
import { Entypo } from '@expo/vector-icons' // Make sure to install @expo/vector-icons

const VerticalDots: React.FC = () => {
  console.log('VerticalDots')

  return (
    <TouchableOpacity style={styles.dots}>
      <Entypo name="dots-three-vertical" size={24} color="black" />
      <Entypo name="dots-three-vertical" size={24} color="black" />
      <Entypo name="dots-three-vertical" size={24} color="black" />
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  dots: {
    padding: 8
  }
})

export default VerticalDots
