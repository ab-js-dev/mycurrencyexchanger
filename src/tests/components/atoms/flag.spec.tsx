import React from 'react'
import { render } from '@testing-library/react-native'
import { Flag } from '../../../components/atoms/flag'

describe('Flag Component', () => {
  it('renders without crashing', () => {
    const { getByTestId } = render(<Flag countryCode="US" />)
    // As we cannot query the inner component directly, we check for the existence of any element.
    // This is a very basic test confirming that the component renders.
    expect(getByTestId).toBeTruthy()
  })

  // Optional: Snapshot test
  it('matches snapshot', () => {
    const tree = render(<Flag countryCode="US" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
