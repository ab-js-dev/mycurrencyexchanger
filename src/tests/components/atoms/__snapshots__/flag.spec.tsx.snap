// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flag Component matches snapshot 1`] = `
<Image
  source={
    {
      "uri": "https://flagcdn.com/w80/us.png",
    }
  }
  style={
    [
      {
        "height": 25,
        "width": 40,
      },
      undefined,
    ]
  }
/>
`;
